src/accel/RunAction.cc

```diff
@@ -59,12 +83,122 @@ void RunAction::EndOfRunAction(const G4Run*) {}
 void RunAction::build_core_params()
 {
     G4AutoLock lock(&mutex);
-    if (false)
+    if (params_->params)
     {
         // Some other thread constructed params between the thread-unsafe check
         // and this thread-safe check
         return;
     }
+
+    celeritas::GeantImporter load_geant_data(
+        G4TransportationManager::GetTransportationManager()
+            ->GetNavigatorForTracking()
+            ->GetWorldVolume());
+
+    auto imported = load_geant_data();
+    CELER_LOG_LOCAL(info) << "loaded data: "
+                          << (imported ? "success" : "failure");
```

Comment:
The log message was good for debugging, but at this point it should be an assertion I think.